server/game/GameActions/DeckSearchAction.ts

```diff
@@ -20,6 +20,12 @@ export class DeckSearchAction extends PlayerAction {
         cardCondition: () => true
     };
 
+    getProperties(context: AbilityContext, additionalProperties = {}): DeckSearchProperties {
+        let properties = super.getProperties(context, additionalProperties) as DeckSearchProperties;
+        properties.reveal = properties.reveal || properties.cardCondition.toString() !== this.defaultProperties.cardCondition.toString();
```

Comment:
This seems really ugly - what are you trying to do here?

Reply:
This is to

if the `reveal:` property has not been set then set `reveal:` to be `true` if `cardCondition:` is not the default (i.e. it has been set) otherwise it will be `false`

The idea is if there is a card condition then you would have to reveal it, but you can still override the implementation by explicitly setting `reveal:`.

The ugly bit is checking whether `cardCondition:` has been set - there may be a better way of doing that?

Reply:
It will override a `reveal: false` - is that what you meant to do? If you want to check whether cardCondition has been set then I think you should not set it in defaultProperties....

Reply:
Yes you are right.  I didn't particularly want to not set cardCondition to () => true as  a default otherwise you have to check it in multiple places to see if it is defined - that seems more messy to me.  I've pushed a new version that hopefully makes the best of both worlds.